core/dbt/parser/manifest.py

```diff
@@ -1121,6 +1158,49 @@ def _process_refs_for_metric(manifest: Manifest, current_project: str, metric: P
         manifest.update_metric(metric)
 
 
+def _process_metrics_for_node(manifest: Manifest, current_project: str, node: Union[ManifestNode, ParsedMetric]):
+    """Given a manifest and a node in that manifest, process its metrics"""
+    for metric in node.metrics:
+        target_metric: Optional[Union[Disabled, ParsedMetric]] = None
+        target_metric_name: str
+        target_metric_package: Optional[str] = None
+
+        if len(metric) == 1:
+            target_metric_name = metric[0]
+        elif len(metric) == 2:
+            target_metric_package, target_metric_name = metric
+        else:
+            raise dbt.exceptions.InternalException(
+                f"Metric references should always be 1 or 2 arguments - got {len(metric)}"
+            )
+
+        # Resolve_ref
+        target_metric = manifest.resolve_metric(
+            target_metric_name,
+            target_metric_package,
+            current_project,
+            node.package_name,
+        )
+
+        if target_metric is None or isinstance(target_metric, Disabled):
+            # This may raise. Even if it doesn't, we don't want to add
+            # this node to the graph b/c there is no destination node
+            node.config.enabled = False
+            # TODO : Different error message?
+            invalid_ref_fail_unless_test(
+                node,
+                target_metric_name,
+                target_metric_package,
+                disabled=(isinstance(target_metric, Disabled)),
+            )
+
+            continue
+
+        target_metric_id = target_metric.unique_id
+
+        node.depends_on.nodes.append(target_metric_id)
```

Comment:
As an aside, I'm not really happy with stuffing everything into depends_on.nodes, even things that aren't in the 'nodes' dictionary. But I guess that's water under the dam :)

Reply:
huh! I think I can figure out a way to add a new property (`.metrics`) under `depends_on`. Very happy to do that - let me know if you'd prefer it!

Reply:
I think we already put sources there too though...